internal/controllers/reconciliation/crud_test.go

```diff
@@ -702,59 +702,6 @@ func TestDisableUpdates(t *testing.T) {
 	assert.Equal(t, "baz", obj.Data["foo"])
 }
 
-// TestOrphanedCompositionDeletion proves that compositions can be deleted when their synthesizer is missing.
-func TestOrphanedCompositionDeletion(t *testing.T) {
```

Comment:
The logic that made this test pass would be... tricky to get working after this change. That said, I don't personally really care about the behavior at least until some other refactoring is done. Open to commenting out the test with a TODO if folks would prefer, but ultimately I don't think it's the right place to test this behavior anyway.

Reply:
I'm okay with deleting this and opening up an issue that points to this PR if we care enough about the behavior to test it after the refactor.